cf_xarray/accessor.py

```diff
@@ -630,15 +630,12 @@ def check_results(names, k):
         else:
             stdnames = set(_get_with_standard_name(obj, k))
             objcoords = set(obj.coords)
-            stdvars = stdnames - objcoords
-            if "coords" not in skip:
-                check_results(stdnames, k)
-                successful[k] = bool(stdnames)
-                coords.extend(stdnames & objcoords)
-            else:
-                check_results(stdvars, k)
-                successful[k] = bool(stdvars)
-            varnames.extend(stdvars)
+            if "coords" in skip:
+                stdnames -= objcoords
```

Comment:
How about this, exactly the same as the origin with an extra if statement?

Reply:
Perfect.